runtime/skiplist.c

```diff
@@ -71,43 +71,42 @@ void caml_skiplist_init(struct skiplist * sk)
 int caml_skiplist_find(struct skiplist * sk, uintnat key, uintnat * data)
 {
   int i;
-  struct skipcell * e, * f;
+  struct skipcell ** e, * f;
 
-  e = (struct skipcell *) sk;
+  e = sk->forward;
   for (i = sk->level; i >= 0; i--) {
     while (1) {
-      f = e->forward[i];
-      if (f == NULL || f->key >= key) break;
-      e = f;
+      f = e[i];
+      if (f == NULL || f->key > key) break;
+      if (f->key == key) {
+        *data = f->data;
+        return 1;
+      }
+      e = f->forward;
     }
   }
-  e = e->forward[0];
-  if (e != NULL && e->key == key) {
-    *data = e->data;
-    return 1;
-  } else {
-    return 0;
-  }
+  return 0;
 }
 
 int caml_skiplist_find_below(struct skiplist * sk, uintnat k,
                              uintnat * key, uintnat * data)
 {
   int i;
-  struct skipcell * e, * f;
+  struct skipcell ** e, * f, * last = NULL;
 
-  e = (struct skipcell *) sk;
+  e = sk->forward;
   for (i = sk->level; i >= 0; i--) {
     while (1) {
-      f = e->forward[i];
+      f = e[i];
       if (f == NULL || f->key > k) break;
-      e = f;
+      last = f;
+      e = f->forward;
     }
   }
-  if (e == (struct skipcell *) sk) {
+  if (!last) {
```

Comment:
I think OCaml's hous style is more on the explicit side:
```suggestion
  if (last != NULL) {
```

Reply:
That would be `last == NULL`...  I can live with `!last`.